src/psyclone/psyir/nodes/call.py

```diff
@@ -593,3 +600,172 @@ def _location_txt(node):
+                        )
+
+                ret_arg_idx_list.append(call_arg_idx)
+                routine_argument_list[call_arg_idx] = None
+                continue
+
+            #
+            # Next, we handle all named arguments
+            #
+            arg_name = self.argument_names[call_arg_idx]
+            named_arg_found = False
+            routine_arg_idx = None
+            for routine_arg_idx, routine_arg in enumerate(
+                                routine_argument_list):
+                routine_arg: DataSymbol
+
+                # Check if argument was already processed
+                if routine_arg is None:
+                    continue
+
+                if arg_name == routine_arg.name:
+                    # TODO #759: If optional is used, it's an unsupported
+                    # Fortran type and we need to use the following workaround
+                    # Once this issue is resolved, simply remove this if branch
+                    if not isinstance(
+                                routine_arg.datatype,
+                                UnsupportedFortranType):
+                        if call_arg.datatype != routine_arg.datatype:
+                            raise CallMatchingArgumentsNotFound(
+                                f"Argument type mismatch of call argument "
+                                f"'{call_arg}' and routine argument "
+                                f"'{routine_arg}'"
+                            )
+
+                    ret_arg_idx_list.append(routine_arg_idx)
+                    named_arg_found = True
+                    break
+
+            if not named_arg_found:
+                # It doesn't match => Raise exception
+                raise CallMatchingArgumentsNotFound(
+                    f"Named argument '{arg_name}' not found"
+                )
+
+            routine_argument_list[routine_arg_idx] = None
+
+        #
+        # Finally, we check if all left-over arguments are optional arguments
+        #
+        for routine_arg in routine_argument_list:
+            routine_arg: DataNode
+
+            if routine_arg is None:
+                continue
+
+            # TODO #759: Optional keyword is not yet supported in psyir.
+            # Hence, we use a simple string match.
+            if ", OPTIONAL" in str(routine_arg.datatype):
+                continue
+
+            raise CallMatchingArgumentsNotFound(
+                f"Argument '{routine_arg.name}' in subroutine"
+                f" '{routine.name}' not handled"
+            )
+
+        return ret_arg_idx_list
+
+    def get_callee(
+                self,
+                ret_arg_match_list: List[int] = None,
+                check_matching_arguments: bool = True):
+        '''
+        Searches for the implementation(s) of the target routine for this Call
+        including argument checks.
+
+        :param ret_arg_match_list: List in which the matching argument
+            indices will be returned
```

Comment:
Please document the `check_matching_arguments` argument.

Reply:
OK.